addon/components/cp-panel-body/template.hbs

```diff
@@ -1,10 +1,12 @@
 {{#if shouldAnimate}}
 
-  {{#liquid-if isOpen use='crossFade'}}
-    <div class="cp-Panel-body-inner">
-      {{yield}}
-    </div>
-  {{/liquid-if}}
+  {{#if (macroDependencySatisfies 'liquid-fire' '*')}}
+    {{#liquid-if isOpen use='crossFade'}}
+      <div class="cp-Panel-body-inner">
+        {{yield}}
+      </div>
+    {{/liquid-if}}
+  {{/if}}
```

Comment:
Instead of this, can we change this

https://github.com/adopted-ember-addons/ember-collapsible-panel/blob/e6bc55477761df5d57de5981c9d97c5ffb5c632c/addon/components/cp-panel-body/component.js#L4-L10

To something like:

```js
import { macroCondition, dependencySatisfies, importSync } from '@embroider/macros';
import { ensureSafeComponent } from '@embroider/util';

let LiquidIf, hasRealLiquidIf;
if (macroCondition(dependencySatisfies('liquid-fire', '*'))) {
  LiquidIf = importSync('liquid-fire/components/liquid-if');
  hasRealLiquidIf = true;
} else {
  hasRealLiquidIf = false;
}

export default Component.extend({
  layout,

  classNames: ['cp-Panel-body'],
  classNameBindings: ['isOpen:cp-is-open'],

  // ideally we'd use a getter here, but that requires Ember 3.16+
  'liquid-if': computed(function() {
      assert('ember-collapsible-panel\'s cp-panel-body component requires liquid-fire to be present if `shouldAnimate` is used', hasRealLiquidIf);
    
     return ensureSafeComponent(LiquidIf);
  }),
});
```

Then change the invocation in the template to:

```suggestion
  {{#this.liquid-if isOpen use='crossFade'}}
    <div class="cp-Panel-body-inner">
      {{yield}}
    </div>
  {{/this.liquid-if}}
```

That lets us remove the `treeForApp` completely (and likely remove the `dependencyChecker` service and the `config/environment` stuff).

Reply:
I like this, I will add an example of this to the macros package docs as I think doing something like this (conditionally wrapping a component) will be a common thing.